pkgs/development/python-modules/scienceplots/default.nix

```
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  matplotlib,
+}:
+
+buildPythonPackage rec {
+  pname = "SciencePlots";
+  version = "2.1.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-2NGX40EPh+va0LnCZeqrWWCU+wgtlxI+g19rwygAq1Q=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [ matplotlib ];
+
+  pythonImportsCheck = [ "scienceplots" ];
+
```

```suggestion
  # no tests
  doCheck = false;
```
We highly recommend that you enable the test.
Please explicitly disable it with comments if you cannot run it.

> Make sure the tests are enabled using for example [pytestCheckHook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook) and, in the case of libraries, are passing for all interpreters. If certain tests fail they can be disabled individually. Try to avoid disabling the tests altogether. In any case, when you disable tests, leave a comment explaining why.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

---

We should note that the practice of disabling empty tests almost only has a positive effect on Rust, since it will recompile anyway. In other cases, running the relevant hooks does not take much time.

---

The compile time may not change, but just adding this one line saves us from having to check for the existence of a test. This is a manual task that can happen many times, so it saves an incredible amount of time :)